### PR TITLE
add an option to completely disable default property source loading

### PR DESCRIPTION
**Pull request by graemerocher**
This adds the ability to disable loading any default property sources if the user wants to control completely what property sources are included

### CODE REVIEW

**Comment by dstepanov** [comment]
I think the alternative to this should be setting `overrideConfigLocations` to nothing and disabling system and env property sources.

File: inject/src/main/java/io/micronaut/context/DefaultApplicationContextBuilder.java [hunk 1]
```diff
@@ -87,6 +88,17 @@ public boolean isAllowEmptyProviders() {
         return allowEmptyProviders;
     }
 
+    @Override
+    public ApplicationContextBuilder enableDefaultPropertySources(boolean areEnabled) {
```
Comment (sdelamo) [inline inject/src/main/java/io/micronaut/context/DefaultApplicationContextBuilder.java:93]:
```suggestion
    @NonNull 
    public ApplicationContextBuilder enableDefaultPropertySources(boolean areEnabled) {
```

**Review by sdelamo — CHANGES_REQUESTED** [review]
Can you add documentation?

**Comment by sdelamo** [comment]
What is the goal of this PR? Faster startup times when you don't want to lose time reading property sources on start-up? Should we add a member to the `@MicronautTest`annotation ? `@MicronautTest(enableDefaultPropertySources = false)`

**Comment by graemerocher** [comment]
It isn't about startup time. Some teams do not want to load configuration from any other location other than the ones they specify manually via `builder.propertySources(..)` for security reasons. In addition for Micronaut for Spring and the starter we don't actually need Micronaut to load the property sources because these are already loaded by Spring so I think it will be valuable for that integration as well

**Comment by graemerocher** [comment]
> Can you add documentation?

The documentation is the javadoc? Don't think this needs specific documentation in the user guide.

**Comment by graemerocher** [comment]
On second though, I think docs explaining the behaviour would be good so I added some

**Comment by graemerocher** [comment]
> I think the alternative to this should be setting `overrideConfigLocations` to nothing and disabling system and env property sources.

Think that is more work than needed, this adds a simply way to achieve that

**Comment by graemerocher** [comment]
> What is the goal of this PR? Faster startup times when you don't want to lose time reading property sources on start-up? Should we add a member to the `@MicronautTest`annotation ? `@MicronautTest(enableDefaultPropertySources = false)`

we could but that will require a downstream change in MicronautTest that would need to be done at a later date, but like I said the goal is not faster startup.

**Review by sdelamo — APPROVED** [review]

**Comment by sdelamo** [comment]
@dstepanov do you have any concerns merging this?

**Review by dstepanov — APPROVED** [review]

**Pull request merged**